### PR TITLE
deprecated @material-ui/core prop-type updates

### PR DESCRIPTION
**Pull request by matthoffner**
## Related Issue

#2935 

## Description

Prop type updates from @material-ui/core deprecated prop types:

* Updates onChangePage  to onPageChange
* Updates onChangeRowsPerPage to onRowsPerPageChange

### CODE REVIEW

**Comment by matthoffner** [comment]
Hi @mbrn this is a PR to fix #2935 with the latest @material-ui/core updates in 4.12.0 causing prop type errors. It would be great if we can get a release out soon with this, thanks!

**Review by aokoruyucu — APPROVED** [review]

**Review by kaledibif — APPROVED** [review]

**Comment by ssp5zone** [comment]
That's weird. mat-ui removed a deprecation in a minor release! That could _(and will)_ break stuff for people if mat-ui dependency is not patch limited.

Anyways, thanks @matthoffner; was going to make this change until I saw your PR.

**Comment by srikanthj09** [comment]
Hi,

can we get this fix release asap,

**Comment by chelproc** [comment]
For those who can't wait for this PR to be released, I wrote a small `TablePagination` patch.

```tsx
import MaterialTable, { MaterialTableProps } from 'material-table';
import { TablePagination, TablePaginationProps } from '@material-ui/core';

function PatchedPagination(props: TablePaginationProps) {
  const {
    ActionsComponent,
    onChangePage,
    onChangeRowsPerPage,
    ...tablePaginationProps
  } = props;

  return (
    <TablePagination
      {...tablePaginationProps}
      // @ts-expect-error onChangePage was renamed to onPageChange
      onPageChange={onChangePage}
      onRowsPerPageChange={onChangeRowsPerPage}
      ActionsComponent={(subprops) => {
        const { onPageChange, ...actionsComponentProps } = subprops;
        return (
          // @ts-expect-error ActionsComponent is provided by material-table
          <ActionsComponent
            {...actionsComponentProps}
            onChangePage={onPageChange}
          />
        );
      }}
    />
  );
}
```

Then you can use this like:

```tsx
<MaterialTable
  components={{
    Pagination: PatchedPagination,
  }}
/>
```

**Comment by ap705** [comment]
Hey, I'm also waiting for this fix to be merged! Could somebody approve it by any chance? Thanks !

**Comment by Beta-7** [comment]
From what i gathered from the gitter chat the project has been abandoned and replaced by https://github.com/material-table-core/core

**Comment by manjuprem12** [comment]
This is blocking us from releasing the latest version of our product. would appreciate this merge the earliest.

**Comment by matthoffner** [comment]
Hello @mbrn could you prioritize reviewing and releasing this to keep this project working with latest material ui changes.

This PR is scoped to fixing a small bug change only, as specifically stated in the README. Thanks.

**Comment by matthoffner** [comment]
Thanks @Beta-7 for the info about a new library. I figured since this was pinned to the maintainer's profile it was a project they wished to maintain but after nearly a month of no response I think its time to move on.

**Comment by arshkkk** [comment]
> For those who can't wait for this PR to be released, I wrote a small `TablePagination` patch.
> 
> ```tsx
> import MaterialTable, { MaterialTableProps } from 'material-table';
> import { TablePagination, TablePaginationProps } from '@material-ui/core';
> 
> function PatchedPagination(props: TablePaginationProps) {
>   const {
>     ActionsComponent,
>     onChangePage,
>     onChangeRowsPerPage,
>     ...tablePaginationProps
>   } = props;
> 
>   return (
>     <TablePagination
>       {...tablePaginationProps}
>       // @ts-expect-error onChangePage was renamed to onPageChange
>       onPageChange={onChangePage}
>       onRowsPerPageChange={onChangeRowsPerPage}
>       ActionsComponent={(subprops) => {
>         const { onPageChange, ...actionsComponentProps } = subprops;
>         return (
>           // @ts-expect-error ActionsComponent is provided by material-table
>           <ActionsComponent
>             {...actionsComponentProps}
>             onChangePage={onPageChange}
>           />
>         );
>       }}
>     />
>   );
> }
> ```
> 
> Then you can use this like:
> 
> ```tsx
> <MaterialTable
>   components={{
>     Pagination: PatchedPagination,
>   }}
> />
> ```

This quick fix is increasing bundle size

**Pull request closed without merging**
